Automattic/jetpack
# JS Components: Convert Gridicon component to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `Gridicon` component to TypeScript.
* It also converts the `Gridicon` stories to TS

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `Gridicon` component work as expected
* Check the consumers of `Gridicon` and related components and confirm that they work as expected
* Check for any regression


<img width="757" alt="Screenshot 2022-07-13 at 4 02 03 PM" src="https://user-images.githubusercontent.com/18226415/178722478-a51a28a6-d21f-47db-af0f-185a15f87823.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202276314264000
